alphagov/paas-cf
# [#154249348] Use ACM for public-facing TLS certs

[#154249348 Use ACM for public-facing TLS certs](https://www.pivotaltracker.com/story/show/154249348)

What?
----

Following on from the spike (#142501669) we now want to proceed with the roll-out of Amazon Certificate Manager (ACM) for the public-facing certs.

In this PR we add the logic to use the ACM certificates using the terraform  ACM certificate datasource to automatically discover them. We also add
a temporary commit with logic to delete the old certificates, which can be reverted in the future.

Dependencies
------------

**IMPORTANT** We added the logic of creating the ACM certificates in alphagov/paas-bootstrap#129. That PR must be merged and applied first.


How to review
---------------

 1. Apply alphagov/paas-bootstrap#129
 2. Run this pipeline.
 3. Check that the certificates have been deleted:

```
DEPLOY_ENV=hector
aws iam list-server-certificates | grep ${DEPLOY_ENV}
aws s3 cp s3://gds-paas-${DEPLOY_ENV}-state/cf-certs.tfstate - # it shall fail
aws s3 cp s3://gds-paas-${DEPLOY_ENV}-state/cf-certs.tar.gz  - | tar -tvzf -  # It shall not include apps or system
```

You can try the code after reverting the temporary commit that deletes the old certs:
 1. Create a tmp branch, revert the last commit and push
 2. Rerun the pipeline.

Next steps.
---------

After merge, wait some time and revert the temporary commit.

Who?
----

Anyone but @keymon or @bandesz
